docs/apis/ou.yaml

```diff
@@ -0,0 +1,262 @@
+                      name: "Frontend Team"
+                      description: "Handles UI/UX work"
+                      parent: "afc77cfd-620b-4cf0-a31c-7377b0ea8902"
+                      sub-organization-units: []
+
+    post:
+      tags:
+        - organization-units
+      summary: Create a new organization unit
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateOrganizationUnitRequest'
+            examples:
+              root-ou:
+                summary: Root organization unit
+                value:
+                  name: "Engineering"
+                  description: "Engineering Unit"
+                  parent: null
+              nested-ou:
+                summary: Nested organization unit
+                value:
+                  name: "Frontend Team"
+                  description: "Handles UI/UX work"
+                  parent: "afc77cfd-620b-4cf0-a31c-7377b0ea8902"
+      responses:
+        '201':
+          description: Organization unit created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrganizationUnit'
+              examples:
+                root-ou:
+                  summary: Root organization unit
+                  value:
+                    id: "afc77cfd-620b-4cf0-a31c-7377b0ea8902"
+                    name: "Engineering"
+                    description: "Engineering Unit"
+                    parent: null
+                nested-ou:
+                  summary: Nested organization unit
+                  value:
+                    id: "4d4cb83a-9bb8-4fd7-970e-36b65f9c8ea4"
+                    name: "Frontend Team"
+                    description: "Handles UI/UX work"
+                    parent: "afc77cfd-620b-4cf0-a31c-7377b0ea8902"
+
+  /organization-units/{id}:
+    get:
+      tags:
+        - organization-units
+      summary: Get a organization unit by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Organization unit details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrganizationUnit'
+              example:
+                id: "afc77cfd-620b-4cf0-a31c-7377b0ea8902"
+                name: "Engineering Team"
+                description: "Engineering unit that handles all engineering tasks"
+                parent: null
+                users:
+                  - "3f9d5c8e-3c56-4f09-a1db-bb3d5e3f4b5d"
+                  - "9f4bc2ee-8d1e-4e21-a3b4-17d9b612e320"
+                  - "20eb225a-e8c3-4ccf-9d47-b659e26014d7"
+                groups:
+                  - "c2a4f62d-7c79-46f2-b2de-8e8cb4f1c456"
+                  - "8a8e4fa2-2e55-478d-bf0f-4a2ddaf9ee7d"
```

Comment:
Do we need to return users and groups with this API?

Reply:
Whats the alternate you propose?

Reply:
IMO we should have separate APIs to fetch users and groups. It can have organization and/ or ou as path or query params. Otherwise when the user count increases this API response will be heavy and probably require pagination.